src/test/java/api/TenantActivationResourceTests.java

```diff
@@ -0,0 +1,66 @@
+package api;
+
+import static api.support.matchers.EventTypeMatchers.isItemCheckedInEventType;
+import static api.support.matchers.EventTypeMatchers.isItemCheckedOutEventType;
+import static api.support.matchers.EventTypeMatchers.isItemDeclaredLostEventType;
+import static api.support.matchers.EventTypeMatchers.isLoanDueDateChangedEventType;
+import static api.support.matchers.PubSubRegistrationMatchers.isValidPublishersRegistration;
+import static org.folio.HttpStatus.HTTP_CREATED;
+import static org.folio.HttpStatus.HTTP_INTERNAL_SERVER_ERROR;
+import static org.folio.HttpStatus.HTTP_NO_CONTENT;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+import org.folio.circulation.support.http.client.Response;
+import org.junit.Before;
+import org.junit.Test;
+
+import api.support.APITests;
+import api.support.fakes.FakePubSub;
+
+public class TenantActivationResourceTests extends APITests {
+
+  @Before
+  public void init() {
+    FakePubSub.setFailPubSubRegistration(false);
+    FakePubSub.setFailPubSubUnregistering(false);
+  }
+
+  @Test
+  public void tenantActivationFailsWhenCannotRegisterWithPubSub() {
+    FakePubSub.setFailPubSubRegistration(true);
+
+    Response response = tenantAPIFixture.postTenant();
+    assertThat(response.getStatusCode(), is(HTTP_INTERNAL_SERVER_ERROR.toInt()));
+  }
+
+  @Test
+  public void tenantActivationSucceedsWhenCanRegisterInPubSub() {
+    Response response = tenantAPIFixture.postTenant();
+    assertThat(response.getStatusCode(), is(HTTP_CREATED.toInt()));
+    assertThat(FakePubSub.getCreatedEventTypes().size(), is(4));
+    assertThat(FakePubSub.getRegisteredPublishers().size(), is(1));
+    assertThat(FakePubSub.getRegisteredSubscribers().size(), is(1));
```

Comment:
There is no subscriber in the messaging descriptor, why do we expect one?

Reply:
PubSub clients makes a call for subscribers anyway. Though I've noticed there's a ticket to change this behavior and it doesn't make sense to rely on it, so this is removed.

Reply:
Thank you!